blueflood-core/src/test/java/com/rackspacecloud/blueflood/service/LocatorFetchRunnableTest.java

```diff
@@ -0,0 +1,366 @@
+package com.rackspacecloud.blueflood.service;
+
+import com.rackspacecloud.blueflood.io.AstyanaxReader;
+import com.rackspacecloud.blueflood.rollup.Granularity;
+import com.rackspacecloud.blueflood.rollup.SlotKey;
+import com.rackspacecloud.blueflood.threading.SizedExecutorService;
+import com.rackspacecloud.blueflood.types.Locator;
+import org.junit.After;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+import org.mockito.Matchers;
+import org.mockito.invocation.InvocationOnMock;
+import org.mockito.stubbing.Answer;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.RejectedExecutionException;
+
+import static org.mockito.Mockito.*;
+
+public class LocatorFetchRunnableTest {
+
+
+    ScheduleContext scheduleCtx;
+    SlotKey destSlotKey;
+    ExecutorService rollupReadExecutor;
+    SizedExecutorService rollupWriteExecutor;
+    ExecutorService enumValidatorExecutor;
+    AstyanaxReader astyanaxReader;
+
+    LocatorFetchRunnable lfr;
+
+    RollupExecutionContext executionContext;
+    RollupBatchWriter rollupBatchWriter;
+
+    List<Locator> locators;
+
+    @Before
+    public void setUp() throws IOException {
+
+        Configuration.getInstance().init();
+
+        this.scheduleCtx = mock(ScheduleContext.class);
+        this.destSlotKey = SlotKey.of(Granularity.FULL, 0, 0);
+        this.rollupReadExecutor = mock(ExecutorService.class);
+        this.rollupWriteExecutor = mock(SizedExecutorService.class);
+        this.enumValidatorExecutor = mock(ExecutorService.class);
+        this.astyanaxReader = mock(AstyanaxReader.class);
+
+        this.lfr = new LocatorFetchRunnable(scheduleCtx,
+                destSlotKey, rollupReadExecutor, rollupWriteExecutor,
+                enumValidatorExecutor, astyanaxReader);
+
+        executionContext = mock(RollupExecutionContext.class);
+        rollupBatchWriter = mock(RollupBatchWriter.class);
+
+        locators = getTypicalLocators();
+
+        Configuration.getInstance().setProperty(CoreConfig.ENABLE_HISTOGRAMS, "false");
+    }
+
+    @After
+    public void tearDown() throws IOException {
+        Configuration.getInstance().init();
+    }
+
+    List<Locator> getTypicalLocators() {
+
+        final Locator locator1 = Locator.createLocatorFromPathComponents("tenant1", "a", "b", "c");
+        final Locator locator2 = Locator.createLocatorFromPathComponents("tenant2", "a", "b", "x");
+        final Locator locator3 = Locator.createLocatorFromPathComponents("tenant3", "d", "e", "f");
+        final List<Locator> locators = new ArrayList<Locator>() {{
+            add(locator1);
+            add(locator2);
+            add(locator3);
+        }};
+        return locators;
+    }
+
+    @Test
+    public void getLocatorsReturnsLocators() {
```

Comment:
Is this test that compelling?  If I understand it correctly, essentially ```getLocators()``` just calls ```astyanaxReader.getLocatorsToRollup()```.  Is that really worth testing?  Am I missing something?  The ```expected``` and ```actual``` are the same class since we are using mockito, right?